docs/old-reference-guide/modules/axon-framework-commands/pages/command-handlers.adoc

```diff
@@ -1,153 +1,215 @@
+        if (cmd.amount() <= 0) {
             throw new IllegalArgumentException("amount <= 0");
         }
-        if (cmd.getAmount() > remainingValue) {
+        if (cmd.amount() > remainingValue) {
             throw new IllegalStateException("amount > remaining value");
         }
-        apply(new CardRedeemedEvent(id, cmd.getTransactionId(), cmd.getAmount()));
+        appender.append(new CardRedeemedEvent(id, cmd.transactionId(), cmd.amount()));
+    }
+
+    // Event sourcing handler to update state
+    @EventSourcingHandler
+    public void on(CardRedeemedEvent event) {
+        this.remainingValue -= event.amount();
     }
-    // omitted event sourcing handlers
 }
 
 ----
 
-The Command objects, `IssueCardCommand` and `RedeemCardCommand`, which `GiftCard` handles have the following format:
+The Command objects, `IssueCardCommand` and `RedeemCardCommand`, which `GiftCard` handles can be defined as simple records (Java 16+) or traditional classes:
 
 [source,java]
 ----
-import org.axonframework.modelling.command.TargetAggregateIdentifier;
-
+// Using Java records (Java 16+)
+public record IssueCardCommand(
+    String cardId,
+    int amount
+) {}
+
+public record RedeemCardCommand(
+    String cardId,
+    String transactionId,
+    int amount
+) {}
+
+// Or using traditional classes
 public class IssueCardCommand {
 
-    @TargetAggregateIdentifier
     private final String cardId;
-    private final Integer amount;
+    private final int amount;
 
-    public IssueCardCommand(String cardId, Integer amount) {
+    public IssueCardCommand(String cardId, int amount) {
         this.cardId = cardId;
         this.amount = amount;
     }
-    // omitted getters, equals/hashCode, toString functions
-}
-
-public class RedeemCardCommand {
-
-    @TargetAggregateIdentifier
-    private final String cardId;
-    private final String transactionId;
-    private final Integer amount;
 
-    public RedeemCardCommand(String cardId, String transactionId, Integer amount) {
-        this.cardId = cardId;
-        this.transactionId = transactionId;
-        this.amount = amount;
-    }
-    // omitted getters, equals/hashCode, toString functions
+    public String cardId() { return cardId; }
+    public int amount() { return amount; }
 }
-
 ----
 
-The `cardId` present in both commands is the reference to a `GiftCard` instance and thus is annotated with the `@TargetAggregateIdentifier` annotation.
-Commands that create an Aggregate instance do not need to identify the target aggregate identifier, as there is no Aggregate in existence yet.
-It is nonetheless recommended for consistency to annotate the Aggregate Identifier on them as well.
+[NOTE]
+====
+The `cardId` present in both commands is the identifier that will be used to route the command to the correct entity instance.
+For command routing in distributed environments, use the `@RoutingKey` annotation on the field that should be used for routing.
```

Comment:
the @RoutingKey annotation is deprecated as of 5.0, should we reference the @CommandHandler#routingKey attribute or the @TargetEntityId annotation instead?